### PR TITLE
Google Storage index support

### PR DESCRIPTION
**Pull request by tom-butler**
add gs-to-tar,

This tool will find a list of files for a prefix, filter them to find the metadata, and add them to a tarball, for later use in the dc-index-from-tar tool. 

updated dc-index-from-tar to support multiple protocols.

### CODE REVIEW

File: dea/apps/index_from_tar.py [hunk 1]
```diff
@@ -51,7 +52,8 @@ def cli(input_fname,
         verify_lineage,
         ignore_lineage,
         gzip,
-        xz):
+        xz,
+        protocol):
```
Comment (Kirill888) [inline dea/apps/index_from_tar.py]: I think we should also accept just `s3` or `gs` for `--protocol` option.

```python
if not protocol.endswith('://'):
    protocol = protocol + '://'
```

File: setup.py [hunk 2]
```diff
@@ -19,10 +19,12 @@
                       'lmdb',
                       'aiobotocore',
                       'botocore',
+                      'google-cloud-storage'
```
Comment (Kirill888) [inline setup.py]: how big is this dependency, should we consider making that an optional dependency?

File: dea/apps/index_from_tar.py (hunk 1 above)
Comment (tom-butler) [reply dea/apps/index_from_tar.py]: rasterio currently supports the following schems:

```python
SCHEMES = {
    'ftp': 'curl',
    'gzip': 'gzip',
    'http': 'curl',
    'https': 'curl',
    's3': 's3',
    'tar': 'tar',
    'zip': 'zip',
    'file': 'file',
    'oss': 'oss',
    'gs': 'gs',
}
```

Comment (tom-butler) [reply dea/apps/index_from_tar.py]: Done

**Pull request merged**